openlibrary/components/LibraryExplorer.vue

```diff
@@ -18,6 +18,41 @@ import DDC from './LibraryExplorer/ddc.json';
 import LCC from './LibraryExplorer/lcc.json';
 import { recurForEach } from './LibraryExplorer/utils.js';
 
+class FilterState {
+    constructor() {
+        this.filter = '';
+        /** @type { '' | 'true' | 'false' } */
+        this.has_ebook = 'true';
+        /** @type {Array<{name: string, key: string}>} */
+        this.languages = [];
+        this.age = '';
+        this.year = '[1985 TO 9998]';
```

Comment:
This could match strange futuristic dates (which do exist in our db) but otherwise seem fine